// react-native-really-awesome-button/lib/commonjs/constants.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.DEFAULT_WIDTH = exports.DEFAULT_TEXT_SIZE = exports.DEFAULT_TEXT_COLOR = exports.DEFAULT_RELEASE_DELAY = exports.DEFAULT_RAISE_LEVEL = exports.DEFAULT_LINE_HEIGHT = exports.DEFAULT_HORIZONTAL_PADDING = exports.DEFAULT_HEIGHT = exports.DEFAULT_DEBOUNCED_PRESS_TIME = exports.DEFAULT_BORDER_WIDTH = exports.DEFAULT_BORDER_RADIUS = exports.DEFAULT_BACKGROUND_SHADOW = exports.DEFAULT_BACKGROUND_DARKER = exports.DEFAULT_BACKGROUND_COLOR = exports.DEFAULT_BACKGROUND_ACTIVE = exports.DEFAULT_ACTIVITY_COLOR = exports.DEFAULT_ACTIVE_OPACITY = exports.ANIMATED_TIMING_OFF = exports.ANIMATED_TIMING_LOADING = exports.ANIMATED_TIMING_IN = exports.ANIMATED_SPRING_TENSION = exports.ANIMATED_SPRING_FRICTION = exports.ANIMATED_ELASTIC_DURATION = exports.ANIMATED_ELASTIC_BOUNCINESS = void 0;
const ANIMATED_SPRING_FRICTION = 6.75;
exports.ANIMATED_SPRING_FRICTION = ANIMATED_SPRING_FRICTION;
const ANIMATED_SPRING_TENSION = 100;
exports.ANIMATED_SPRING_TENSION = ANIMATED_SPRING_TENSION;
const ANIMATED_ELASTIC_DURATION = 300;
exports.ANIMATED_ELASTIC_DURATION = ANIMATED_ELASTIC_DURATION;
const ANIMATED_ELASTIC_BOUNCINESS = 1.2;
exports.ANIMATED_ELASTIC_BOUNCINESS = ANIMATED_ELASTIC_BOUNCINESS;
const ANIMATED_TIMING_OFF = 100;
exports.ANIMATED_TIMING_OFF = ANIMATED_TIMING_OFF;
const ANIMATED_TIMING_IN = 200;
exports.ANIMATED_TIMING_IN = ANIMATED_TIMING_IN;
const ANIMATED_TIMING_LOADING = 3000;
exports.ANIMATED_TIMING_LOADING = ANIMATED_TIMING_LOADING;
const DEFAULT_RAISE_LEVEL = 4;
exports.DEFAULT_RAISE_LEVEL = DEFAULT_RAISE_LEVEL;
const DEFAULT_ACTIVE_OPACITY = 1;
exports.DEFAULT_ACTIVE_OPACITY = DEFAULT_ACTIVE_OPACITY;
const DEFAULT_RELEASE_DELAY = 100;
exports.DEFAULT_RELEASE_DELAY = DEFAULT_RELEASE_DELAY;
const DEFAULT_HEIGHT = 60;
exports.DEFAULT_HEIGHT = DEFAULT_HEIGHT;
const DEFAULT_WIDTH = null;
exports.DEFAULT_WIDTH = DEFAULT_WIDTH;
const DEFAULT_DEBOUNCED_PRESS_TIME = 0;
exports.DEFAULT_DEBOUNCED_PRESS_TIME = DEFAULT_DEBOUNCED_PRESS_TIME;
const DEFAULT_BORDER_RADIUS = 4;
exports.DEFAULT_BORDER_RADIUS = DEFAULT_BORDER_RADIUS;
const DEFAULT_BORDER_WIDTH = 0;
exports.DEFAULT_BORDER_WIDTH = DEFAULT_BORDER_WIDTH;
const DEFAULT_HORIZONTAL_PADDING = 16;
exports.DEFAULT_HORIZONTAL_PADDING = DEFAULT_HORIZONTAL_PADDING;
const DEFAULT_TEXT_COLOR = '#FFFFFF';
exports.DEFAULT_TEXT_COLOR = DEFAULT_TEXT_COLOR;
const DEFAULT_ACTIVITY_COLOR = '#FFFFFF';
exports.DEFAULT_ACTIVITY_COLOR = DEFAULT_ACTIVITY_COLOR;
const DEFAULT_BACKGROUND_COLOR = '#c0c0c0';
exports.DEFAULT_BACKGROUND_COLOR = DEFAULT_BACKGROUND_COLOR;
const DEFAULT_BACKGROUND_DARKER = '#9f9f9f';
exports.DEFAULT_BACKGROUND_DARKER = DEFAULT_BACKGROUND_DARKER;
const DEFAULT_BACKGROUND_SHADOW = 'rgba(0, 0, 0, 0.15)';
exports.DEFAULT_BACKGROUND_SHADOW = DEFAULT_BACKGROUND_SHADOW;
const DEFAULT_BACKGROUND_ACTIVE = 'rgba(0, 0, 0, 0.15)';
exports.DEFAULT_BACKGROUND_ACTIVE = DEFAULT_BACKGROUND_ACTIVE;
const DEFAULT_LINE_HEIGHT = 20;
exports.DEFAULT_LINE_HEIGHT = DEFAULT_LINE_HEIGHT;
const DEFAULT_TEXT_SIZE = 14;
exports.DEFAULT_TEXT_SIZE = DEFAULT_TEXT_SIZE;
