// react-native-really-awesome-button/lib/commonjs/themed/social/constants.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.YOUTUBE = exports.WHATSAPP = exports.TWITTER = exports.REDDIT = exports.PINTEREST = exports.MESSENGER = exports.LINKEDIN = exports.GITHUB = exports.FACEBOOK = void 0;
const TWITTER = {
  backgroundColor: "#00aced",
  backgroundDarker: "#0096cf"
};
exports.TWITTER = TWITTER;
const MESSENGER = {
  backgroundColor: "#3186f6",
  backgroundDarker: "#2566bc"
};
exports.MESSENGER = MESSENGER;
const FACEBOOK = {
  backgroundColor: "#4868ad",
  backgroundDarker: "#325194"
};
exports.FACEBOOK = FACEBOOK;
const GITHUB = {
  backgroundColor: "#2c3036",
  backgroundDarker: "#060708"
};
exports.GITHUB = GITHUB;
const LINKEDIN = {
  backgroundColor: "#0077b5",
  backgroundDarker: "#005885"
};
exports.LINKEDIN = LINKEDIN;
const WHATSAPP = {
  backgroundColor: "#25d366",
  backgroundDarker: "#14a54b"
};
exports.WHATSAPP = WHATSAPP;
const REDDIT = {
  backgroundColor: "#fc461e",
  backgroundDarker: "#d52802"
};
exports.REDDIT = REDDIT;
const PINTEREST = {
  backgroundColor: "#bd091c",
  backgroundDarker: "#980313"
};
exports.PINTEREST = PINTEREST;
const YOUTUBE = {
  backgroundColor: "#cc181e",
  backgroundDarker: "#ab0d12"
};
exports.YOUTUBE = YOUTUBE;
